src/block.rs

```diff
@@ -170,46 +170,93 @@ impl BlockBuilder {
 
 #[cfg(test)]
 mod tests {
+    use rstest::rstest;
+
     use super::*;
-    use crate::test_utils::{gen_attrs, gen_empty_attrs};
+    use crate::test_utils::{assert_debug_snapshot, gen_attrs, gen_empty_attrs};
 
-    #[test]
-    fn test_block() {
-        let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key2", b"value2", gen_empty_attrs()));
-        let block = builder.build().unwrap();
-        let encoded = block.encode();
-        let decoded = Block::decode(encoded);
-        assert_eq!(block.data, decoded.data);
-        assert_eq!(block.offsets, decoded.offsets);
+    #[derive(Debug)]
+    struct BlockTestCase {
+        name: &'static str,
+        entries: Vec<(&'static [u8], Option<&'static [u8]>)>, // (key, value) with None for tombstone
+        expected_size: Option<usize>,                         // Expected size of the block
+        use_timestamped_attrs: bool, // Flag to use attributes with a timestamp
```

Comment:
Since we're moving to table-driven testing, I think we should test all attributes, not just timestamp. I suggest dropping `use_timestamped_attrs` and `timestamp` and replacing these with an `attrs: RowAttributes` field. Then, add some additional test cases to test not only `ts`, but also `ttl`. I think you won't be able to use `gen_attrs` anymore because `BlockTestCase` is being created in a macro, but that's OK. Should be able to do:

```
    #[case(BlockTestCase {
        name: "block_size_with_attrs",
        entries: vec![
            (b"key1", Some(b"value1")),
            (b"key2", Some(b"value2")),
        ],
        expected_size: Some(73),
        attrs: RowAttributes {
            ts: Some(1),
            ttl: None,
        },
    })]
```

Reply:
Alternatively, you could use `RowEntry` instead of the k/v bytes. That might actually be cleaner, and lets us also test the seq num as well.

Reply:
It will also allow to express merge operands. In general we should stop expressing entries as Optional<Bytes> throughout the codebase, because it doesn't contain full information about what the inner value represents. I've already cleaned that up in most places.

Reply:
> Alternatively, you could use `RowEntry` instead of the k/v bytes. That might actually be cleaner, and lets us also test the seq num as well.

I did a research on how to best incorporate a test case using [RowEntry] Following the code in [block.rs], which internally calls row_codec for encoding, I found that to better test fields like sequence number and timestamp, I need to use the decoding logic from row_codec.

The test utility will help provides output like this:


```
SstRowEntry { key_prefix_len: 0, key_suffix: b"key3key4", seq: 0, expire_ts: Some(0), create_ts: Some(0), value: Value(b"value1") }
SstRowEntry { key_prefix_len: 8, key_suffix: b"", seq: 0, expire_ts: Some(0), create_ts: Some(0), value: Merge(b"value2") }
SstRowEntry { key_prefix_len: 3, key_suffix: b"4key5", seq: 0, expire_ts: None, create_ts: Some(0), value: Tombstone }
```

Having that will help better position to be testable. I noticed also that we compute prefixes by capturing the common prefix between two keys and keeping the suffix. I need a way to reconstruct the full key from the key_prefix_len and key_suffix.